app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/photo/PhotoTaskViewModel.kt

```diff
@@ -63,13 +63,13 @@ constructor(
     requireNotNull(currentTask) { "Photo captured but no task waiting for the result" }
 
     try {
-      val bitmap = bitmapUtil.fromUri(uri)
-      val file = userMediaRepository.savePhoto(bitmap, currentTask)
+      val file = userMediaRepository.savePhotoFromUri(uri, currentTask)
       userMediaRepository.addImageToGallery(file.absolutePath, file.name)
       val remoteFilename = FirebaseStorageManager.getRemoteMediaPath(surveyId, file.name)
-      setValue(PhotoTaskData(remoteFilename))
+
+      withContext(Dispatchers.Main) { setValue(PhotoTaskData(remoteFilename)) }
```

Comment:
As discussed, we can directly update the value from background thread using `postValue` without having to change the thread.